### ISSUE TITLE
Forms can be submitted with validation failing when using `validateOnBlur`

### ISSUE THREAD

**Issue by estephan** [marmelab/react-admin#4181]
**What you were expecting:**

I was expecting that validation would run before I could submit a form.

**What happened instead:**

When submitting a form, a user can bypass validation and save a form that is invalid.

**Steps to reproduce:**

1. Creating the following form:
```
<Create {...props}>
    <SimpleForm validateOnBlur>
      <TextInput source="name" validate={minLength(10)} fullWidth />
    </SimpleForm>
  </Create>
```
2. Type "foo" into the `TextInput` (or any string having length less than 10).
3. Click the "Save" button.

**Related code:**

* Preferably, a CodeSandbox forked from https://codesandbox.io/s/github/marmelab/react-admin/tree/master/examples/simple
* A link to a GitHub repo with the minimal codebase to reproduce the issue

I've just added `validateOnBlur` to the existing `CommentCreate` form. So if you select a Post from the dropdown and then type a string in `Author` that is less than 10 characters and click "Save" you can reproduce the issue.

https://codesandbox.io/s/agitated-feather-r98vi

**Other information:**

**Environment**

* React-admin version: 3.0.4
* Last version that did not exhibit the issue (if applicable): 2.x
* React version: 16.12.0
* Browser: Chrome (v 78.0.3904.108)

**Comment by fzaninotto** [comment]
Reproduced and confirmed, thanks. This is due to the fact that we trigger the form submission on mouse down on the SaveButton, and that's before the blur event starts in the active field. We need to fix that.

**Comment by fzaninotto** [comment]
In fact, I think it's a final-form bug. `validateOnBlur` should not disable the validation on submission, because there are cases like ours where it's creates problems. 

https://github.com/final-form/react-final-form/issues/70#issuecomment-567441111

**Comment by JulienMattiussi** [comment]
@estephan I think this issue has been fixed in ra 3.2.3 with the changes realised in form submission button.

This fork of your codesandbox with updated packages doesn't have the problem.
https://codesandbox.io/s/cool-roentgen-3bw4o

Note : Also, I don't find any impact of validateOnBlur because setting true or false doens't seems to change the behaviour

**Comment by estephan** [comment]
Can confirm this is not fixed 👍
